code/modules/mob/living/carbon/human/species/station/vaurca/vaurca.dm

```diff
@@ -149,6 +149,29 @@
 	var/obj/item/clothing/mask/gas/vaurca/filter/M = new /obj/item/clothing/mask/gas/vaurca/filter(H)
 	H.equip_to_slot_or_del(M, slot_wear_mask)
 
+	if(HAS_TRAIT(H, TRAIT_ORIGIN_HAS_PSIAUG))
+		var/obj/item/organ/A = new /obj/item/organ/internal/augment/psi(H)
+		var/obj/item/organ/external/affected = H.get_organ(A.parent_organ)
+		A.replaced(H, affected)
+	if(HAS_TRAIT(H, TRAIT_ORIGIN_HAS_KLAX_IMPLANT))
+		var/obj/item/organ/A = new /obj/item/organ/internal/augment/language/klax(H)
+		var/obj/item/organ/external/affected = H.get_organ(A.parent_organ)
+		A.replaced(H, affected)
+	if(HAS_TRAIT(H, TRAIT_ORIGIN_HAS_CTHUR_IMPLANT))
+		var/obj/item/organ/A = new /obj/item/organ/internal/augment/language/cthur(H)
+		var/obj/item/organ/external/affected = H.get_organ(A.parent_organ)
+		A.replaced(H, affected)
+	if(HAS_TRAIT(H, TRAIT_ORIGIN_HAS_MIKUETZ_IMPLANT))
+		var/obj/item/organ/A = new /obj/item/organ/internal/augment/language/mikuetz(H)
+		var/obj/item/organ/external/affected = H.get_organ(A.parent_organ)
+		A.replaced(H, affected)
+	if(HAS_TRAIT(H, TRAIT_ORIGIN_HAS_ZINO_IMPLANT))
+		var/obj/item/organ/A = new /obj/item/organ/internal/augment/language/zino(H)
+		var/obj/item/organ/external/affected = H.get_organ(A.parent_organ)
+		A.replaced(H, affected)
```

Comment:
This is a horridly hardcoded way of handling this. You need to generalize this whole behaviour somehow. Offloading it to the origin and generalizing it to work with all augments is a start.

Reply:
Not sure why it's asking for a changelog again, but change addressed.